platform/viewer/src/routes/WorkList/WorkList.tsx

```diff
@@ -429,6 +432,34 @@ function WorkList({
     });
   }
 
+  const handleUploadComplete = () => {
+    hide();
+    onRefresh();
+  };
+
+  const handleUploadStarted = () => {
+    show({
+      ...uploadProps,
+      closeButton: false,
+    });
+  };
+
+  const uploadProps = {
+    title: 'Upload files',
+    closeButton: true,
+    shouldCloseOnEsc: false,
+    shouldCloseOnOverlayClick: false,
+    content: DicomUpload.bind(null, {
+      dataSource,
+      onComplete: handleUploadComplete,
+      onStarted: handleUploadStarted,
+    }),
+  };
```

Comment:
I recommend reducing the lines of code with somethign like this, which is more readable to me too

```js
const uploadProps = {
  title: "Upload files",
  closeButton: true,
  shouldCloseOnEsc: false,
  shouldCloseOnOverlayClick: false,
  content: DicomUpload.bind(null, {
    dataSource,
    onComplete: () => {
      hide()
      onRefresh()
    },
    onStarted: () => {
      show({
        ...uploadProps,
        closeButton: false,
      })
    },
  }),
}
```